src/cms/api/site/models/site.js

```diff
@@ -5,4 +5,20 @@
  * to customize this model
  */
 
-module.exports = {};
+const regex = new RegExp("^[a-z0-9/]+(?:-[a-z0-9/]+)*$|^$");
+const slugValidator = (data) => {
```

Comment:
Can we create a shared file instead of having the same code in 4 places?

Reply:
I thought the same thing but faced [this problem](https://forum.strapi.io/t/how-to-structure-reusable-code-to-import-in-various-strapi-files/2435/3).

- Created `utils/helpers.js` under `cms`
- In `helpers.js`
```
const regex = new RegExp("^[a-z0-9/]+(?:-[a-z0-9/]+)*$|^$");
export const slugValidator = (slug) => {
    if (!regex.test(slug)) {
      throw strapi.errors.badRequest('Please enter letters, numbers, hyphens, or slashes for slug. No spaces.');
    }
};
```
- In `page.js`
```
import { slugValidator } from "../../../utils/helpers.js";
```
- Got an error: `SyntaxError: Cannot use import statement outside a module`

Seems like I could use [this configuration](https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/configurations.html#functions) but not sure what to return.

Do you have any ideas?

Reply:
It's actually server-side node code not client-side javascript code.  So I believe you need to use `require` instead of `import`

`const slugValidator = require("../../../utils/helpers.js").slugValidator();`

(I think it's something like that anyway)